mapcreator/building.py

```diff
@@ -127,42 +146,43 @@ def merge(buildstatus, debug = False):
     """
     Merge source map files into a single file.
     """ 
-    combined_file_name = INTERMEDIATE_FILENAME_FORMAT.format('_combined')
-    single_path = path.join(BUILD_DIR, combined_file_name)
-    sourcefiles_as_string = ' '.join(buildstatus.current_files)
-    command = 'gdalwarp {} {}'.format(sourcefiles_as_string, single_path)
-    call_command(command, buildstatus, debug)
-    buildstatus.add_next_file(single_path)
+    if buildstatus.current_files:
+        combined_file_name = INTERMEDIATE_HEIGHT_FILENAME_FORMAT.format('_combined')
+        single_path = path.join(BUILD_DIR, combined_file_name)
+        sourcefiles_as_string = ' '.join(buildstatus.current_files)
+        command = 'gdalwarp {} {}'.format(sourcefiles_as_string, single_path)
+        call_command(command, buildstatus, debug)
+        buildstatus.add_next_file(single_path)
     buildstatus.next()
 
 def translate(buildstatus, debug = False):
     for cf in buildstatus.current_files:
-        outpath = get_output_path(cf, OUTPUT_FILE_EXTENSION)
-        metapath = get_output_path(cf, METADATA_FILE_EXTENSION)
+        outpath = get_output_path(cf, HEIGHT_OUTPUT_FILE_EXTENSION)
+        metapath = get_output_path(cf, HEIGHT_METADATA_FILE_EXTENSION)
 
-        command = 'gdal_translate -of {} {} {}'.format(OUTPUT_FORMAT, cf, outpath)
+        command = 'gdal_translate -of {} {} {}'.format(HEIGHT_OUTPUT_FORMAT, cf, outpath)
         call_command(command, buildstatus, debug)
 
-        finalized_metaname = FINAL_METADATA_FORMAT.format(buildstatus.get_metaindex())
+        finalized_metaname = FINAL_HEIGHT_METADATA_FORMAT.format(buildstatus.get_metaindex())
         finalized_metapath = path.join(FINALIZED_DIR, finalized_metaname)
         rename(metapath, finalized_metapath)
 
         buildstatus.add_next_file(outpath)
         buildstatus.add_result_file(finalized_metapath)
     buildstatus.next()
 
-
 def finalize(buildstatus, debug = False):
-    if buildstatus.current_files == buildstatus.original_files:
```

Comment:
Any particular reason for getting rid of this? This was a safeguard to make sure that the original files that were given for mapcreator wouldn't be modified

Reply:
The logic currently goes that if there are not any files for a certain stage of heightmap processing, the current files list ends up empty. So it will only equal original_files if the original_files list is empty. The check was changed to enable the build to run using only XML or satellite files (without heightmap files or with heightmap files not coinciding with the window).

Reply:
What if every action before finalize throws an exception at the actions first line of code for some reason? In that case we'll have the original files as current files and they'll end up getting renamed. The build should be able to run without height files even with this check in - the exception ends up getting caught in do_build's iteration of actions.
Might be cleaner to just return instead of throwing an exception, though, as catching an exception sets has_errors and this isn't explicitly an error...

Reply:
Okay, valid point. Basically that would occur if the user had added files that could not be found during build. Returned the requested check. I also improved running the build with nonexistent files in different categories and nonexistent window.